src/dds/rsdds-serializable.cpp

```diff
@@ -0,0 +1,171 @@
+#include <src/core/device-interface.h>
+
+#include <rsutils/json.h>
+#include <set>
+
+using json = nlohmann::json;
+
+
+namespace librealsense {
+
+
+dds_serializable::dds_serializable()
+{
+}
+
+
+std::vector< uint8_t > dds_serializable::serialize_json() const
+{
+    serialized_utilities::json_preset_writer writer;
+    writer.set_device_info( get_serializable_device() );
+
+    // Set of options that should not be written
+    std::set< rs2_option > const options_to_ignore{
+        RS2_OPTION_FRAMES_QUEUE_SIZE,  // Internally added and is not an option we need to record/load
+        RS2_OPTION_REGION_OF_INTEREST  // The RoI is temporary, and we don't load it
+    };
+
+    for( auto p_sensor : get_serializable_sensors() )
+    {
+        std::string const sensor_name = p_sensor->get_info( RS2_CAMERA_INFO_NAME );
+        for( auto o : p_sensor->get_supported_options() )
+        {
+            auto & opt = p_sensor->get_option( o );
+            if( opt.is_read_only() )
+                continue;
+            if( options_to_ignore.find( o ) != options_to_ignore.end() )
+                continue;
+
+            auto val = opt.get_value();
+            writer.write_param( sensor_name + '/' + get_string( o ), val );
+        }
+    }
+
+    auto str = writer.to_string();
+    return std::vector< uint8_t >( str.begin(), str.end() );
+}
+
+
+void dds_serializable::load_json( const std::string & json_content )
+{
+    serialized_utilities::json_preset_reader reader( json_content );
+
+    // Verify if device information in preset file is compatible with the connected device.
+    reader.check_device_info( get_serializable_device() );
+    auto const sensors = get_serializable_sensors();
+
+    // Do the visual preset first, as it sets the basis on top of which we apply the regular controls
+    {
+        auto const visual_preset_subkey = '/' + get_string( RS2_OPTION_VISUAL_PRESET );
+        for( auto const p_sensor : sensors )
+        {
+            std::string const sensor_name = p_sensor->get_info( RS2_CAMERA_INFO_NAME );
+            auto const key = sensor_name + visual_preset_subkey;
+            auto it = reader.find( key );
+            if( it == reader.end() )
+                continue;
+
+            try
+            {
+                auto & opt = p_sensor->get_option( RS2_OPTION_VISUAL_PRESET );
+                opt.set_value( it.value() );
+                LOG_DEBUG( "loaded '" << key << "' value " << it.value() );
+            }
+            catch( std::exception const & e )
+            {
+                LOG_ERROR( "Failed to load '" << key << "' (value " << it.value() << "): " << e.what() );
+                throw;
+            }
+        }
+    }
+
+    // Create a set of options we have to change
+    std::map< std::string /*key*/, std::string /*error*/ > keys_to_change;
+    {
+        // Set of options that should not be set in the loop
+        std::set< rs2_option > const options_to_ignore{
```

Comment:
Please make `options_to_ignore` common for serialize and load

Reply:
They're not the same, though some elements are usually common. But OK will try, maybe with a static function...